lib/common/output_xml.c

```diff
@@ -448,6 +454,11 @@ pcmk__output_xml_add_node(pcmk__output_t *out, xmlNodePtr node) {
     CRM_ASSERT(priv != NULL);
     CRM_ASSERT(node != NULL);
 
+    if (!pcmk__str_eq(out->fmt_name, "xml", pcmk__str_none) &&
+        !pcmk__str_eq(out->fmt_name, "html", pcmk__str_none)) {
+        return;
+    }
+
```

Comment:
This makes me realize we're silently relying on private_data_t to start with the same members for xml and html. We need to document or change that

Reply:
I don't really like them having to be the same either, but I can't think of a better way to do it aside from duplicating all the XML functions for the HTML side.  So maybe a comment is acceptable for the moment.